doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,892 @@
+                type enumeration {
+                    enum ewma {
+                        description "Exponentially Weighted Moving Average algorithm";
+                    }
+                }
+            }
+
+            leaf max_flows {
+                type uint32;
+                description  "Maximum number of flows that can be maintained per ARS profile.";
+            }
+
+            leaf sampling_interval {
+                type uint32;
+                description  "Sampling interval in microseconds for quality measure computation.";
+            }
+
+            leaf past_load_min_value {
+                type uint16;
+                description "Past load min value.";
+            }
+
+            leaf past_load_max_value {
+                type uint16;
+                description "Past load max value.";
+            }
+
+            leaf past_load_weight {
+                type uint16;
+                description "Past load weight.";
+            }
+
+            leaf future_load_min_value {
+                type uint16;
+                description "Future load min value.";
+            }
+            leaf future_load_max_value {
+                type uint16;
+                description "Future load max value.";
+            }
+
+            leaf future_load_weight {
+                type uint16;
+                description "Future load weight.";
+            }
+
+            leaf current_load_min_value {
+                type uint16;
+                description "Current load min value.";
+            }
+
+            leaf current_load_max_value {
+                type uint16;
+                description "Current load max value.";
+            }
+
+            leaf ipv4_enable {
+                type boolean;
+                description "Whether ARS is enabled over IPv4 packets";
+            }
+
+            leaf ipv6_enable {
+                type boolean;
+                description "Whether ARS is enabled over IPv6 packets";
+            }
+        }
+        /* end of list ARS_PROFILE_LIST */
+    }
+    /* end of container ARS_PROFILE */
+```
+
+##### ARS_INTERFACE
+
+```
+    container ARS_INTERFACE {
+
+        list ARS_INTERFACE_LIST {
+            description  "List of interfaces participating in ARS";
+            key "if_name";
+
+            leaf if_name {
+                type leafref {
+                    path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name";
+                }
+                description "ARS-enabled interface name";
+            }
+
+            leaf scaling_factor {
+                type uint32;
+                default "10000";
+                description "This factor used to normalize load measurements across ports with different speeds.";
+            }
+
+        }
+        /* end of list ARS_INTERFACE_LIST */
+    }
+    /* end of container ARS_INTERFACE */
+```
+
+##### ARS_NEXTHOP_GROUP
+
+```
+    container ARS_NEXTHOP_GROUP {

```

Comment:
ARS_PREFIX could be a better name since key is an IP Prefix and multiple prefixes can share the same nexthop group.